cli/azd/pkg/tools/python/python.go

```diff
@@ -119,13 +128,25 @@ func (cli *PythonCli) CreateVirtualEnv(ctx context.Context, workingDir, name str
 	return nil
 }
 
-func pythonExe() string {
+func checkPath() (string, bool, error) {
+	var found bool
+	var err error
 	if runtime.GOOS == "windows" {
-		if _, err := osexec.LookPath("py"); err != nil {
-			return "python"
+		// py for https://peps.python.org/pep-0397
+		pyString := [2]string{"py", "python"}
+
+		for _, py := range pyString {
+			found, err = tools.ToolInPath(py)
+			if found && err == nil {
+				return py, found, nil
+			}
 		}
-		return "py" // https://peps.python.org/pep-0397
+		return "", found, err
```

Comment:
I think this ends up returning `nil` error if not found? And so the consumers that only check if `err != nil` won't correctly detect when the tool isn't in PATH?